pkg/extension/sumologicextension/credentials.go

```diff
@@ -0,0 +1,181 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sumologicextension
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/extension/sumologicextension/api"
+	"go.uber.org/zap"
+)
+
+// Authentication is an interface to get collector authentication data
+type CredsGetter interface {
+	CheckCollectorCredentials() bool
+	GetStoredCredentials() (api.OpenRegisterResponsePayload, error)
+	RegisterCollector() (api.OpenRegisterResponsePayload, error)
+}
+
+// Authenticator is a common structure for both types of getting collector credentials
+// stored in file and registration new collector.
+type credsGetter struct {
+	conf   *Config
+	logger *zap.Logger
+}
+
+// CheckCollectorCredentials checks if collector credentials can be found in path
+// configured in the config.
+func (cr credsGetter) CheckCollectorCredentials() bool {
+	filenameHash, err := hash(cr.conf.CollectorName)
+	if err != nil {
+		return false
+	}
+	path := path.Join(cr.conf.CollectorCredentialsPath, filenameHash)
+	if _, err := os.Stat(path); err != nil {
+		return false
+	}
+	return true
+}
+
+// GetStoredCredentials retrieves collector credentials stored in local file system and then decrypts it
+// using hashed collector name as passphrase.
+func (cr credsGetter) GetStoredCredentials() (api.OpenRegisterResponsePayload, error) {
+	filenameHash, err := hash(cr.conf.CollectorName)
+	if err != nil {
+		return api.OpenRegisterResponsePayload{}, err
+	}
+
+	path := path.Join(cr.conf.CollectorCredentialsPath, filenameHash)
+	creds, err := os.Open(path)
+	if err != nil {
+		return api.OpenRegisterResponsePayload{}, err
+	}
+
+	defer creds.Close()
+	encryptedCreds, err := ioutil.ReadAll(creds)
+	if err != nil {
+		return api.OpenRegisterResponsePayload{}, err
+	}
+
+	collectorCreds, err := decrypt(encryptedCreds, cr.conf.CollectorName)
+	if err != nil {
+		return api.OpenRegisterResponsePayload{}, err
+	}
+
+	var credentialsInfo api.OpenRegisterResponsePayload
+	if err = json.Unmarshal(collectorCreds, &credentialsInfo); err != nil {
+		return api.OpenRegisterResponsePayload{}, err
+	}
+
+	return credentialsInfo, nil
+}
+
+// RegisterCollector registers new collector using registration API and returns collector credentials.
+func (cr credsGetter) RegisterCollector() (api.OpenRegisterResponsePayload, error) {
```

Comment:
```suggestion
func (cr credsGetter) RegisterCollector(ctx context.Context) (api.OpenRegisterResponsePayload, error) {
```

I believe we can easily add context to this func (and use it in the request creation)